test/NuGet.Core.Tests/NuGet.DependencyResolver.Core.Tests/TrackerTests.cs

```diff
@@ -0,0 +1,188 @@
+        Assert.False(tracker.IsDisputed(itemA2));
+        Assert.False(tracker.IsDisputed(itemB1));
+
+        tracker.Track(itemA2);
+
+        Assert.True(tracker.IsDisputed(itemA1));
+        Assert.True(tracker.IsDisputed(itemA2));
+        Assert.False(tracker.IsDisputed(itemB1));
+
+        tracker.Track(itemB1);
+
+        Assert.True(tracker.IsDisputed(itemA1));
+        Assert.True(tracker.IsDisputed(itemA2));
+        Assert.False(tracker.IsDisputed(itemB1));
+
+        tracker.Clear();
+
+        Assert.False(tracker.IsDisputed(itemA1));
+        Assert.False(tracker.IsDisputed(itemA2));
+        Assert.False(tracker.IsDisputed(itemB1));
+    }
+
+    [Fact]
+    public void GetDisputes()
+    {
+        var itemA1 = MakeItem("A", 1);
+        var itemA2 = MakeItem("A", 2);
+        var itemB1 = MakeItem("B", 1);
+
+        var tracker = new Tracker<string>();
+
+        Assert.Empty(tracker.GetDisputes(itemA1));
+        Assert.Empty(tracker.GetDisputes(itemA2));
+        Assert.Empty(tracker.GetDisputes(itemB1));
+
+        tracker.Track(itemA1);
+
+        Validate(itemA1, new[] { itemA1 });
+        Validate(itemA2, new[] { itemA1 });
+        Assert.Empty(tracker.GetDisputes(itemB1));
+
+        tracker.Track(itemA2);
+
+        Validate(itemA1, new[] { itemA1, itemA2 });
+        Validate(itemA2, new[] { itemA1, itemA2 });
+        Assert.Empty(tracker.GetDisputes(itemB1));
+
+        tracker.Clear();
+
+        Assert.Empty(tracker.GetDisputes(itemA1));
+        Assert.Empty(tracker.GetDisputes(itemA2));
+        Assert.Empty(tracker.GetDisputes(itemB1));
+
+        void Validate(GraphItem<string> item, GraphItem<string>[] expected)
+        {
+            IEnumerable<GraphItem<string>> disputes = tracker.GetDisputes(item);
+
+            Assert.True(new HashSet<GraphItem<string>>(disputes).SetEquals(expected));
+        }
+    }
+
+    [Fact]
+    public void IsAmbiguous()
+    {
+        var itemA1 = MakeItem("A", 1);
+        var itemB1 = MakeItem("B", 1);
+
+        var tracker = new Tracker<string>();
+
+        Assert.False(tracker.IsAmbiguous(itemA1));
+        Assert.False(tracker.IsAmbiguous(itemB1));
+
+        tracker.MarkAmbiguous(itemA1);
+
+        Assert.True(tracker.IsAmbiguous(itemA1));
+        Assert.False(tracker.IsAmbiguous(itemB1));
+
+        tracker.MarkAmbiguous(itemB1);
+
+        Assert.True(tracker.IsAmbiguous(itemA1));
+        Assert.True(tracker.IsAmbiguous(itemB1));
+
+        tracker.Clear();
+
+        Assert.False(tracker.IsAmbiguous(itemA1));
+        Assert.False(tracker.IsAmbiguous(itemB1));
+    }
+
+    [Fact]
+    public void IsBestVersion()
+    {
+        var itemA1 = MakeItem("A", 1);
+        var itemA2 = MakeItem("A", 2);
+        var itemA3 = MakeItem("A", 3);
+        var itemB1 = MakeItem("B", 1);
+
+        var tracker = new Tracker<string>();
+
+        // NOTE this is strange behavior. An item that was never tracked is reported as the best version.
```

Comment:
Isn't this a consequence of the API behaving as a GetOrAdd?

Reply:
That's right. That's how the prior code was written, and it's been preserved here. I called it out here in a comment as the test looks strange to me. I'm not sure it's safe to change the behaviour, so left it as-is.

Reply:
I think it makes to not change the behavior, but I'm not sure if the comment here adds much. 
IMO, given that the behavior is get or add, would we better off to have a comment there?

Reply:
Sure, we could add a comment to `IsBestVersion` too, explaining the strange behaviour. I like tests to read clearly though on their own, hence the comment here. To me this test reads like:

```c#
List<int> numbers = new();
numbers.Contains(1); // true?
```

Reply:
Sure, I don't mind a version of the test comment if we document the behavior in the class itself. That's where people are most likely to read about the behavior anyway.s

Reply:
Pushed an update. Do we have to get everyone's approval again now?